File: testers/grpc/java/generated-and-checked-in/com/example/grpc/Inner.java
package com.example.grpc;

import com.google.protobuf.CodedInputStream;
import com.google.protobuf.CodedOutputStream;
import com.google.protobuf.WireFormat;
import io.grpc.MethodDescriptor.Marshaller;
import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.io.InputStream;

public record Inner(Integer value, String description) {
  public Inner withValue(Integer value) {
    return new Inner(value, description);
  }

  public Inner withDescription(String description) {
    return new Inner(value, description);
  }

  public static Marshaller<Inner> MARSHALLER =
      new Marshaller<Inner>() {
        @Override
        public InputStream stream(Inner value) {
          var bytes = new byte[value.getSerializedSize()];
          var cos = CodedOutputStream.newInstance(bytes);
          try {
            value.writeTo(cos);
            cos.flush();
          } catch (IOException e) {
            throw new RuntimeException(e);
          }
          return new ByteArrayInputStream(bytes);
        }

        @Override
        public Inner parse(InputStream stream) {
          try {
            return Inner.parseFrom(CodedInputStream.newInstance(stream));
          } catch (IOException e) {
            throw new RuntimeException(e);
          }
        }
      };

  public static Inner parseFrom(CodedInputStream input) throws IOException {
    Integer value = 0;
    String description = "";
    while (!input.isAtEnd()) {
      var tag = input.readTag();
      if (WireFormat.getTagFieldNumber(tag) == 1) {
        value = input.readInt32();
      } else if (WireFormat.getTagFieldNumber(tag) == 2) {
        description = input.readString();
      } else {
        input.skipField(tag);
      }
      ;
    }
    ;
    return new Inner(value, description);
  }

  public Integer getSerializedSize() {
    Integer size = 0;
    size = size + CodedOutputStream.computeInt32Size(1, this.value());
    size = size + CodedOutputStream.computeStringSize(2, this.description());
    return size;
  }

  public void writeTo(CodedOutputStream output) throws IOException {
    output.writeInt32(1, this.value());
    output.writeString(2, this.description());
  }
}

File: testers/grpc/scala/generated-and-checked-in/com/example/grpc/Inner.scala
package com.example.grpc

import com.google.protobuf.CodedInputStream
import com.google.protobuf.CodedOutputStream
import com.google.protobuf.WireFormat
import io.grpc.MethodDescriptor.Marshaller
import java.io.ByteArrayInputStream
import java.io.IOException
import java.io.InputStream
import java.lang.RuntimeException

case class Inner(
  value: Int,
  description: String
) {
  @throws[IOException]
  def writeTo(output: CodedOutputStream): Unit = {
    output.writeInt32(1, this.value)
    output.writeString(2, this.description)
  }

  def getSerializedSize: Int = {
    var size: Int = 0
    size = size + CodedOutputStream.computeInt32Size(1, this.value)
    size = size + CodedOutputStream.computeStringSize(2, this.description)
    return size
  }
}

object Inner {
  given marshaller: Marshaller[Inner] = {
    new Marshaller[Inner] {
      override def stream(value: Inner): InputStream = {
        val bytes = Array.ofDim[Byte](value.getSerializedSize)
        val cos = CodedOutputStream.newInstance(bytes)
        try {
          value.writeTo(cos)
          cos.flush()
        } catch {
          case e: IOException => throw new RuntimeException(e)
        } 
        return new ByteArrayInputStream(bytes)
      }
      override def parse(stream: InputStream): Inner = {
        try {
          return Inner.parseFrom(CodedInputStream.newInstance(stream))
        } catch {
          case e: IOException => throw new RuntimeException(e)
        } 
      }
    }
  }

  @throws[IOException]
  def parseFrom(input: CodedInputStream): Inner = {
    var value: Int = 0
    var description: String = ""
    while (!input.isAtEnd()) {
      val tag = input.readTag()
      if (WireFormat.getTagFieldNumber(tag) == 1) { value = input.readInt32() }
      else if (WireFormat.getTagFieldNumber(tag) == 2) { description = input.readString() }
      else { input.skipField(tag) }
    }
    return new Inner(value, description)
  }
}